packages/server/src/index.ts

```diff
@@ -89,6 +121,96 @@ export default class Server {
 
     this.serverAndClient.addMethod(method, handler);
   }
+
+  /**
+   * Set the instance [[currencies]] object
+   *
+   * @param params.walletCurrencies - the list of all the currencies handled by
+   * the wallet
+   * @param params.allowedCurrenciesFilters - this list of allowed currencies
+   * filters for a specific Wallet APP, extracted from it's `categories`
+   * manifest field.
+   * These filters must be glob
+   * https://en.wikipedia.org/wiki/Glob_(programming)
+   */
+  setCurrencies({
+    walletCurrencies,
+    allowedCurrenciesFilters,
+  }: {
+    walletCurrencies: Currency[];
+    allowedCurrenciesFilters: "*" | string[];
+  }): void {
+    // If the Wallet APP has access to all available currencies (i.e: no restriction)
+    if (allowedCurrenciesFilters === "*") {
+      this.currencies = walletCurrencies;
+      return;
+    }
+
+    const filteredCurrencies = walletCurrencies.filter((currency) => {
+      // Test if a specific currency is allowed based on provided filters
+      return allowedCurrenciesFilters.some((filter) => {
+        const re = globStringToRegex(filter);
+        return currency.id.match(re);
+      });
+    });
+
+    this.currencies = filteredCurrencies;
+  }
+
+  /**
+   * Get the instance [[currencies]] object
+   *
+   * @returns The list of currencies available for the Wallet APP communicating
+   * with this server instance.
+   */
+  getCurrencies(): Currency[] | undefined {
+    return this.currencies;
+  }
+
+  /**
+   * Set the instance [[accounts]] object
+   *
+   * @param params.walletAccounts - the list of all the accounts handled by
+   * the wallet
+   * @param params.allowedCurrenciesFilters - this list of allowed currencies
+   * filters for a specific Wallet APP, extracted from it's `categories`
+   * manifest field.
+   * These filters must be glob
+   * https://en.wikipedia.org/wiki/Glob_(programming)
+   */
+  setAccounts({
+    walletAccounts,
+    allowedCurrenciesFilters,
+  }: {
+    walletAccounts: Account[];
+    allowedCurrenciesFilters: "*" | string[];
+  }): void {
+    // If the Wallet APP has access to all available currencies (i.e: no restriction)
+    if (allowedCurrenciesFilters === "*") {
+      this.accounts = walletAccounts;
+      return;
+    }
+
+    const filteredAccounts = walletAccounts.filter((account) => {
+      // Test if a specific account is allowed based on provided filters
+      return allowedCurrenciesFilters.some((filter) => {
+        const re = globStringToRegex(filter);
+        return account.currency.match(re);
+      });
```

Comment:
Same question about the regex

Reply:
Fixed by 4f295d89f66564b1d3376c34748101259a6bfadb